### PR TITLE
drivers: ps2: Microchip XEC PS2 add MEC172x support

### PR DESCRIPTION
**Pull request by jvasanth1**
Update the Microchip XEC PS2 driver to support MEC172x. Also add pinctrl support.

### CODE REVIEW

File: drivers/ps2/Kconfig.xec [hunk 1]
```diff
@@ -19,6 +20,7 @@ config PS2_XEC_0
 
 config PS2_XEC_1
 	bool "PS2_XEC_1"
+	depends on SOC_SERIES_MEC1501X
```
Comment (MaureenHelm) [inline drivers/ps2/Kconfig.xec:23]: `PS2_XEC_0` and `PS2_XEC_1` both need to be removed

File: boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig [hunk 2]
```diff
@@ -18,5 +18,6 @@ CONFIG_UART_CONSOLE=y
 CONFIG_ADC=y
 CONFIG_I2C=y
 CONFIG_ESPI=y
+CONFIG_PS2=y
```
Comment (MaureenHelm) [inline boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig:21]: This whole file needs to be cleaned up to include only the drivers needed for hello_world

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please split the cleanup changes and MEC172x changes into separate commits. There are also some twister failures that need to be addressed.

File: drivers/ps2/Kconfig.xec (hunk 1 above)
Comment (jvasanth1) [reply drivers/ps2/Kconfig.xec:23]: MEC15xx has 2 PS2 controllers and MEC17xx has only one. Hence the` depends on SOC_SERIES_MEC1501X` is not applicable `PS2_XEC_0`

File: boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig (hunk 2 above)
Comment (jvasanth1) [reply boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig:21]: okay, will remove adc, i2c, espi, ps2, spi, spi_async

**Comment by jvasanth1** [comment]
> Please split the cleanup changes and MEC172x changes into separate commits. There are also some twister failures that need to be addressed.

split into separate commits

File: boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig (hunk 2 above)
Comment (jvasanth1) [reply boards/arm/mec172xevb_assy6906/mec172xevb_assy6906_defconfig]: including only drivers for hello_world

File: drivers/ps2/Kconfig.xec (hunk 1 above)
Comment (MaureenHelm) [reply drivers/ps2/Kconfig.xec:23]: The number of instances is defined in the SoC-level devicetrees, and after the refactoring in this PR to use `DT_INST_FOREACH_STATUS_OKAY`, the driver isn't using these Kconfig symbols anymore.

Comment (jvasanth1) [reply drivers/ps2/Kconfig.xec]: removed `PS2_XEC_0 `and `PS2_XEC_1 `symbols

**Comment by jvasanth1** [comment]
Any inputs on the twister error?

It seems to be a BOUNDS_CHECK_BYPASS_MITIGATION error. Not sure how to debug this.

```
warning: BOUNDS_CHECK_BYPASS_MITIGATION (defined at kernel/Kconfig:762) was assigned the value 'y'
but got the value 'n'.
```

**Comment by jvasanth1** [comment]
Hi @MaureenHelm I fixed the build errors. Please help to approve.

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: boards/arm/mec1501modular_assy6885/pinmux.c [hunk 3]
```diff
@@ -280,7 +280,7 @@ static int board_pinmux_init(const struct device *dev)
 	pinmux_pin_set(portb, MCHP_GPIO_073, MCHP_GPIO_CTRL_MUX_F1);
 #endif
 
-#ifdef CONFIG_PS2_XEC_0
+#if DT_NODE_HAS_STATUS(DT_INST(0, microchip_xec_ps2), okay)
```
[the remark this answers is not in the document]
Comment (jvasanth1) [reply boards/arm/mec1501modular_assy6885/pinmux.c]: fixed accordingly

**Review by albertofloyd — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**